README.ideal.md

```diff
@@ -0,0 +1,145 @@
+_Ni-Mh — batteries included environments with Nickel inside_.
+
+Managing a project's development environment involves configuring a lot of tools and occasionnally getting to communicate with each other: One (or several) package manager, a CI system, a service manager (to run that local postgresql database without which you can't test anything), custom utility scripts that end up spawning everywhere, etc.
+Ni-Mh (called “nickel-nix” or “Nixel” at the moment, but I really like Ni-Mh) aims at being your main entrypoint for managaing all these different tools, so that you can get:
+
+1. A unified configuration framework for all of these
+2. A powerful and ergonomic language to allow you to easily abstract over these configurations, with discoverability and early error reporting
+
+## Getting started
+
+To start using `nickel-nix`, you need `Nix` to be installed and configured.
+If it isn't already the case, you can get it with:
+
+```console
+$ curl -L https://nixos.org/nix/install | bash
+# We also need a couple of experimental Nix features
+$ mkdir -p ~/.config/nix
+$ echo 'experimental-features = nix-command flakes' >> ~/.config/nix/nix.conf
+```
+
+Then bootstrap a project with:
+
+```console
+$ nix flake init -t github:nickel-lang/nickel-nix
+# Edit the project file to fit your needs
+$ $EDITOR project.ncl
+# Enter the environment
+$ nix develop
+```
+
+## Managing your environment with Nixel
+
+### Dependencies
+
+Nixel can be used to declare the dependencies for your project.
```

Comment:
Should we also provide a way for user to not write `flake.nix`, but rather declare all dependencies in `project.ncl` and let nixel manage the flake?

Reply:
Yes, that's the intent for the 80% case. But should we mention it explicitly given that the premise is “Nixel will manage your tools for you”?

Reply:
Yeah, I guess we don't need to add this here.